src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -158,6 +163,104 @@ class SchemaService(private val options: Neo4jOptions, private val jobId: String
     struct
   }
 
+  def countForNode(): Long = try {
+    /*
+     * we try to leverage the count store in order to have the faster response possible
+     * https://neo4j.com/developer/kb/fast-counts-using-the-count-store/
+     * so in this scenario we have some limitations given the fact that we get the min
+     * for the sequence of counts returned
+     */
+    val query = "CALL apoc.meta.stats() yield labels RETURN labels"
+    val map = session.run(query).single()
+      .asMap()
+      .asScala
+      .get("labels")
+      .getOrElse(Collections.emptyMap())
+      .asInstanceOf[util.Map[String, Long]].asScala
+    map.filterKeys(k => options.nodeMetadata.labels.contains(k))
+      .values.min
+  } catch {
+    case e: ClientException =>
+      e.code match {
+        case "Neo.ClientError.Procedure.ProcedureNotFound" => {
+          // TODO get back to Cypher DSL
+          options.nodeMetadata.labels
+            .map(label => {
+              val query = s"""MATCH (${Neo4jUtil.NODE_ALIAS}:$label)
+                             |RETURN count(${Neo4jUtil.NODE_ALIAS}) AS count
+                             |""".stripMargin
+              val count = session.run(query).single().get("count")
+              if (count.isNull) 0L else count.asLong()
+            })
+            .min
+        }
+      }
+  }
+
+  def countForRelationship(): Long = try {
```

Comment:
These count methods, please think about failure methods.  In some of these APOC calls you are doing you are coupling down to the string format that APOC returns in a particular field.  Remembering that APOC will get put into product at some point, and the string format of return columns isn't promised, this code has some potential to break at some point in the future due to a new published APOC, just as an example of what can go wrong.

Suppose the user specifies paralleism of 5, and *everything goes wrong* with the count feature, and there's no way to obtain a valid count.   What happens then?    My vote would be for a debug message, plus reverting to reading into one partition (with no need for counts)

Reply:
there's the extra possibility that the user just regular screws up the count query.   Say I specify the count query like this:

```
MATCH (p:Person)-[:BOUGHT]->(pr:Product)
RETURN count(pr) - 10 as count
```

Doesn't really matter how or why, but the count is wrong.

In this case, you might try to fetch one extra "empty" partition.  If the user says 5 partitions, try to fetch 6, where the 6th partition has one record (SKIP whatever LIMIT 1).   If you get nothing, that's good.  If you get an actual record here, this is a warning that the user has messed up the partition.   Another warning is if any of your partitions end up empty, then chances are good the user messed up the count query.

Reply:
just allow to specifiy a literal count value in the property

Reply:
It's always good to overfetch for stuff like that in SDN we used to pull pages with pagesize+1 so we could say "there is a next page"

Reply:
Just to be sure should we over-fetch always or only if the user provides the query count right?

Reply:
> just allow to specifiy a literal count value in the property

the user can define a query like this `RETURN 50 AS count` is it fine? Or we allow to simply pass `.option("query.count", "50")`?

Reply:
I added `.option("query.count", "50")` as valid option